backends/dpdk/dpdkContext.cpp

```diff
@@ -451,6 +466,199 @@ void DpdkContextGenerator::addMatchTables(Util::JsonArray* tablesJson) {
+    auto hwblk = new LookupHwBlocks();
+    hwblk->name = "MOD";
+    // hwblk->id = MOD;
+    hwblk->hw_interface = "CP_CHANNEL";
+    hwblk->hw_resource = resname;
+    hwblk->hw_resource_id = 7;
+    exactMVLut->matchAttributes = new LookupMatchAttributes();
+    exactMVLut->matchAttributes->hardware_blocks.push_back(hwblk);
+    UpdateMatchKeys(exactMVLut, type0);
+    UpdateImmediateFields(exactMVLut, type1);
+    contextLutTables.push_back(exactMVLut);
+}
+
+void DpdkContextGenerator::outputImmediateField(Util::JsonArray* immFields,
+        ImmediateFields* immfld) {
+    auto* immField = new Util::JsonObject();
+    immField->emplace("param_name", immfld->param_name);
+    immField->emplace("param_handle", immfld->param_handle);
+    immField->emplace("dest_start", immfld->dest_start);
+    immField->emplace("bitwidth", immfld->bitwidth);
+    immFields->append(immField);
+}
+
+void DpdkContextGenerator::outputKeys(Util::JsonObject* keyJsonObj,
+    std::vector<struct MatchKeyField*> &keylist, bool isLut) {
+    auto* keyJsons = new Util::JsonArray();
+    keyJsonObj->emplace("match_key_fields", keyJsons);
+
+    for (auto keyInfo : keylist) {
+        auto* keyJson = new Util::JsonObject();
+        keyJsons->append(keyJson);
+        keyJson->emplace("name", keyInfo->name);
+        if (keyInfo->isInstanceName) {
+            keyJson->emplace("instance_name", keyInfo->instanceName);
+        }
+        if (keyInfo->isFieldName) {
+            keyJson->emplace("field_name", keyInfo->fieldName);
+        }
+        if (keyInfo->matchType == exact) {
+            keyJson->emplace("match_type", "exact");
+        } else if (keyInfo->matchType == ternary) {
+            keyJson->emplace("match_type", "ternary");
+        } else if (keyInfo->matchType == lpm) {
+            keyJson->emplace("match_type", "lpm");
+        } else if (keyInfo->matchType == range) {
+            keyJson->emplace("match_type", "range");
+        } else if (keyInfo->matchType == selector) {
+            keyJson->emplace("match_type", "selector");
+        }
+        keyJson->emplace("start_bit", keyInfo->startBit);
+        keyJson->emplace(isLut ? "dest_width" : "bit_width", keyInfo->bitWidth);
+        keyJson->emplace("bit_width_full", keyInfo->bitWidthFull);
+        keyJson->emplace("index", keyInfo->index);
+        keyJson->emplace("position", keyInfo->position);
+    }
+}
+
+void DpdkContextGenerator::outputLutMatchAttributes(Util::JsonObject* matchJsons,
+     struct P4MatchLookupTableInfo* tblInfo) {
+    auto* matchAttr0 = new Util::JsonObject();
+    auto* matchAttrs = new Util::JsonArray();
+    matchJsons->emplace("match_attributes", matchAttr0);
+    matchAttr0->emplace("hardware_blocks", matchAttrs);
+    if (tblInfo->matchAttributes != nullptr) {
+        for (auto hwBlock : tblInfo->matchAttributes->hardware_blocks) {
+            auto* matchAttr = new Util::JsonObject();
+            matchAttr->emplace("name", hwBlock->name);
```

Comment:
Some of these fields may be not applicable for DPDK

Reply:
removed